tests/sentry/options/test_manager.py

```diff
@@ -106,6 +106,29 @@ def test_legacy_key(self):
         assert self.manager.delete("sentry:foo")
         assert self.manager.get("sentry:foo") == ""
 
+    def test_legacy_url_prefix_key_use_db_setting(self):
+        """
+        TODO: Remove when SENTRY_URL_PREFIX is completely deprecated
+        """
+        self.manager.register("system.url-prefix", flags=FLAG_PRIORITIZE_DISK)
+        with patch.object(self.store.cache, "get", return_value="https://sentry.example.com"):
+            with self.settings(SENTRY_OPTIONS={}):
+                assert self.manager.get("system.url-prefix") == "https://sentry.example.com"
+                assert settings.SENTRY_URL_PREFIX == "https://sentry.example.com"
+                assert settings.SENTRY_OPTIONS["system.url-prefix"] == "https://sentry.example.com"
+
+    def test_legacy_url_prefix_use_config_setting(self):
+        """
+        TODO: Remove when SENTRY_URL_PREFIX is completely deprecated
+        """
+        self.manager.register("system.url-prefix", flags=FLAG_PRIORITIZE_DISK)
+        with patch.object(self.store.cache, "get", return_value="https://sentry.example.com"):
+            with self.settings(SENTRY_OPTIONS={"system.url-prefix": "https://sentry.updated.com"}):
+                assert self.manager.get("system.url-prefix") == "https://sentry.updated.com"
+                # SENTRY_URL_PREFIX is set on initialization, then not modified if config settings change
+                assert settings.SENTRY_URL_PREFIX == "http://testserver"
```

Comment:
I don't like that this is different from `SENTRY_OPTIONS["system.url-prefix"]`, but since this is deprecated I don't want to really touch any of the previous logic around this. Also, we always use the `system.url-prefix` if it's provided instead so this seems safe

Reply:
Can we expand the comment to that effect? Suggestion below ...

Reply:
This is set in `test.py`. If you prefer a different value, you could override in `with self.settings()` above.